app/services/migration/migrators/user.rb

```diff
@@ -52,11 +62,36 @@ def unique_validated_trns(ecf_user)
         .uniq
     end
 
+    def unique_emails(ecf_user)
+      emails = ecf_user.npq_profiles.map { |pp| pp.participant_identity.email }
+      emails += ecf_user.npq_applications.map { |app| app.participant_identity.email }
+      emails.map { |email| email.to_s.downcase.strip }.compact.uniq
+    end
+
+    def most_recent_updated_npq_application(ecf_user)
+      ecf_user.npq_applications.order(updated_at: :desc).first
```

Comment:
having a further think about this, maybe it's worth also looking for the user -> profile -> npq application if the npq application is empty here as opposed to what I suggested earlier regarding adding the safe navigation operator, it feels safer 🤔